### DISCUSSION TITLE
Commercial forks of Foxglove 1.x?

### DISCUSSION THREAD

**Discussion by Ext3h** [foxglove/community#974] [General]
Well, less off a question, more of an observation / statement.

A couple of fortune 50 companies had forked Foxglove 1.x quite a while ago and had since kept selling those forks as their own products to customers in premium markets with their own branding. They had also restored / added features which had been defined out of the open source edition.

Probably also worth noting: NONE of them went for MCAP as the backing file storage format. Pretty much the first thing you do is attaching your own data lake / preferred storage format, and then you keep investing into making Foxglove scalable with extremely broad models (e.g. making an 6 digit number of scalar signals accessible for browsing and alike).

Just food for thought what this means for the market position of Foxglove 2.0, as this highly restrictive model now at best serves to frighten off adopters, and puts Foxglove as just yet another mark in a long, long list of proprietary visualization solutions. Most of which most likely even slipped through in the market analysis, as you can't even image how many proprietary visualization frameworks for company internal use exist (even when "only" considering software with an at minimum 4 digits user base).

Cancelling the open source option has just canceled the momentum you had from riding on the popularity of ROS2, which was actually on track as making you the set default option for visualization in a long list of domains.

And that also at a somewhat non-competitive pricing model. For real - putting on-premise hosting in the highest price bracket? That's the most basic requirement for most companies to keep external web-services out of the equation. You don't have the reputation / track record to be trusted with sensitive data yet ;)

And you don't have the manpower either to compete against the forks of your own software at this point, now that you have become actual competitors. (Not like you would had made money from it before, but now it's getting even harder and your stream of market insight is going to dry up, too.)

**Comment by amacneil** [comment]
Appreciate the feedback.

I was of course as disappointed as anyone to reach the conclusion that we need to switch to a commercial model. However, as you pointed out, there are many large companies getting substantial value from Foxglove and not paying us. 

> A couple of fortune 50 companies had forked Foxglove 1.x quite a while ago and had since kept selling those forks as their own products to customers in premium markets with their own branding. 

Unfortunately, the robotics industry is not big enough that we can justify spending the majority of our engineering effort on a product that we give away for free. Companies in other industries have been able to make open source work when they operate in a massive market and can afford for only a small percentage of their users to pay.

> Cancelling the open source option has just canceled the momentum you had from riding on the popularity of ROS2, which was actually on track as making you the set default option for visualization in a long list of domains.

Unfortunately, being the default option for visualization does not help us if we cannot support our development efforts. We're continuing to offer a free plan for small teams (up to 3 users) or academic teams of any size. For larger companies using Foxglove in a commercial setting we ask them to upgrade to a paid plan in order to access 2.x.

**Reply by Ext3h** [reply]
> Unfortunately, the robotics industry is not big enough that we can justify spending the majority of our engineering effort on a product that we give away for free.

Hrm, not sure how to tell you this, but your product had crept out of the robotics industry into far more lucrative markets and with contract volumes which definitely exceed your expectations. Just not with your brand, so you haven't gotten recognition for it, and you missed out on the lucrative customization jobs as well ;) Yet you are even nice enough to advertise for some of those partners who have taken your software into those domains right on your front page.

You are kind of missing the primary selling point too. Your software as a standalone solution / data pool is not actually worth paying for, you are barking up directly at major players such as Microsoft and IBM and co when it comes to data storage solutions for the customers you actually want to focus on, and they will snipe every single larger customer from you.

It had the promise of enriching existing appliances / applications with an universally accessible and yet low (performance) impact visualization. But that's in markets where per-user subscription models / non-perpetual licenses are still frowned upon (partially due to compliance topics), and on-premise - or should I say "on the edge" - hosting is the normal scenario. Also, open source is becoming more and more of a deciding factor due to compliance reasons as well, you'd be surprised how badly those premium customers have been burnt over time by closed source software.

**Comment by martinerk0** [comment]
@Ext3h which companies?  I'm curious :)

@amacneil Why not going gitlab way, and have two versions? 

- **ce** - community edition with basic stuff
- **ee** - enterprise edition with stuff that big corpos use

I only tried foxglove few years ago, but rviz was in the end sufficient for me.

**Reply by Ext3h** [reply]
Try e.g. Nvidia which has based the visualization for their Drive PX platform on Foxglove, except they didn't attribute anything. And yet is selling it as part of their own platform to large automotive corporations. The later do not even know yet that they are Foxglove users, and likewise the Foxglove developers didn't realize that they already were in that market...

We are not even talking about ROS2 or MCAP integration in such a context, Foxglove is being used for the visualization capabilities alone. Which is what I meant with: that EE model with the current or even the previous feature set simply had no relevant selling points.

Except there is now the major risk that those markets will ditch Foxglove in favor of RViz, since that one is actually known by name.

**Reply by amacneil** [reply]
> likewise the Foxglove developers didn't realize that they already were in that market

To be clear, we are fully aware of this usage and have a commercial relationship with NVIDIA.
